azurerm/internal/services/media/media_asset_filter_resource_test.go

```diff
@@ -0,0 +1,201 @@
+package media_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/media/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type AssetFilterResource struct{}
+
+func TestAccAssetFilter_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_media_asset_filter", "test")
+	r := AssetFilterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				check.That(data.ResourceName).Key("name").HasValue("Filter-1"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccAssetFilter_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_media_asset_filter", "test")
+	r := AssetFilterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				check.That(data.ResourceName).Key("name").HasValue("Filter-1"),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccAssetFilter_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_media_asset_filter", "test")
+	r := AssetFilterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				check.That(data.ResourceName).Key("first_quality_bitrate").HasValue("128000"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
```

Comment:
could we add a update test that goes from basic -> complete -> basic?

Reply:
Sure